src/cracking.py

```diff
@@ -0,0 +1,307 @@
+	moduels = ["requests"]
+	offline = False
+
+	# How do we implment the API key?
+
+	def crack(self, hash):
+		try:
+			request = requests.get(
+				f"https://hashes.org/api.php?key={key}&query={hash[0]}", timeout=1
+			).text
+		except:
+			logger.warn("Couldn't connect to hashesorg")
+			return False
+
+		# Check for false positive
+
+		if "null" in request:
+			logger.warn("Couldnt connect to hashesorg")
+			return False
+		if "" == request:
+			logger.warn("Hash seems to be a plain???")
+			return False
+
+		output = request.split('":"')[2].split('","')[0]
+
+		# Check for Hex
+
+		if "$HEX[" in output:
+			return bytearray.fromhex(
+				final_output[5 : len(final_output) - 1]
+			).decode()  # Partions it so that it only contains hex and then decodes it into ASCII
+
+		return output
+
+# Need to support IDS but will wait for format from name-that-hash
+class hashcat:
+
+	supports = set(["md5", "NTLM", ""])
+	moduels = ["subprocess"]
+	offline = True
+
+	# How do we implement the types?? (IDs) and wordlist??
+
+	def crack(self, hash):
+		for possible_type in range(len(ids)):
+			command = f"hashcat64.exe -a 0 -m {ids[possible_type]} {hash[0]} {wordlist}"  # >> Can change for better optimization etc.....
+			# Hashcat64.exe for windows
+			try:
+				sp.check_call(
+					command,
+					shell="True",
+					cwd="/mnt/p/Hashcat"  # << For windows, you must put the hashcat directory for it to work. Also the wordlist must be in the directory or a full path.
+					# stderr=sp.STDOUT
+				)
+			except sp.CalledProcessError as error:
+				if (
+					"'hashcat' is not recognized as an internal or external command"
+					in str(error)
+				):
+					logger.warn("Hashcat not in PATH")
+					return False
+				if "./hashcat.hctune: No such file or directory" in str(error):
+					logger.warn("Read the docs on using windows")
+					return False
+				logger.warn("Hashcat couldn't crack hash")
+				continue
+
+			possible_output = str(
+				sp.check_output(
+					f"{command} --show", shell=True, cwd="/mnt/p/Hashcat"
+				).strip()
+			).replace("'", "")
+
+			if not "No hashes loaded." in possible_output:
+				return possible_output.split(":")[1]
+
+# Do later
+class john:
+	pass
+
+	"""
+
+	supports = set(["","",""])
+	moduels = ["subprocess"]
+
+	def crack(self, hash, quiet):
+	if quiet:
+		john = subprocess.Popen(
+			["john", hash, "--wordlist=" + wordlist], stdout=subprocess.PIPE)
+		output, error = john.communicate()
+		print("Jesus take the wheel! (output not supressed)")
+	
+	else:
+		with open('/tmp/jtr_out.txt', 'w') as output:
+			john = subprocess.Popen(
+			("john", "--fork=4", hash, "--wordlist=" + wordlist,), stdout=output, stderr=STDOUT)
+			print("Output Supressed!")
+			out, error = john.communicate()
+
+	Thanks jabbba, will do this later :P
+
+	"""
+
+# Completed
+class LmRainbowTabels:
+
+	# Ok so bug for this one, it doesnt like any word longer then 7 charcters :*(, also it for some reason puts it ALL in caps wtf?
+
+	supports = set(["lm"])

```

Comment:
none of the hashes in name-that-hash have rainbow table stuff?

Reply:
do u want me to add it

Reply:
rainbow tabels are just a way of cracking the hash, this only supports LM hashs so as long as the LM hash is supported in nth all good